pkg/fixtures/writer_s3.go

```diff
@@ -0,0 +1,103 @@
+package fixtures
+
+import (
+	"context"
+	"github.com/applike/gosoline/pkg/blob"
+	"github.com/applike/gosoline/pkg/cfg"
+	"github.com/applike/gosoline/pkg/mon"
+	"github.com/aws/aws-sdk-go/aws"
+	"os"
+	"path/filepath"
+)
+
+type BlobFixturesSettings struct {
+	BasePath   string
+	ConfigName string
+}
+
+type s3FixtureWriter struct {
+	logger      mon.Logger
+	store       blob.Store
+	batchRunner *blob.BatchRunner
+
+	settings *BlobFixturesSettings
+	purger   *s3Purger
+}
+
+func S3FixtureWriterFactory(settings *BlobFixturesSettings) FixtureWriterFactory {
+	return func(config cfg.Config, logger mon.Logger) (FixtureWriter, error) {
+		store := blob.NewStore(config, logger, settings.ConfigName)
+		br := blob.NewBatchRunner(config, logger)
+		purger := newS3Purger(config, logger, settings)
+
+		return NewS3FixtureWriterWithInterfaces(logger, store, br, settings, purger), nil
+	}
+}
+
+func NewS3FixtureWriterWithInterfaces(logger mon.Logger, store blob.Store, batchRunner *blob.BatchRunner, settings *BlobFixturesSettings, purger *s3Purger) FixtureWriter {
+	return &s3FixtureWriter{
+		logger:      logger,
+		store:       store,
+		settings:    settings,
+		batchRunner: batchRunner,
+		purger:      purger,
+	}
+}
+
+func (s *s3FixtureWriter) Purge() error {
+	return s.purger.purge()
+}
+
+func (s *s3FixtureWriter) Write(*FixtureSet) error {
+	s.store.CreateBucket()
+
+	var files []string
+	err := filepath.Walk(s.settings.BasePath, func(path string, f os.FileInfo, err error) error {
+		if f.IsDir() {
+			return nil
+		}
```

Comment:
I think we should handle it, if there was an issue traversing through the file tree. Also info will be nil if err != nil. See documentation for WalkFunc.
```suggestion
		if err != nil {
			return err
		}

		if f.IsDir() {
			return nil
		}
```

Reply:
Applied.

Reply:
I can not see it on my just-now fetched/pulled branch. Please check again!

Reply:
you can now